pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -463,7 +463,10 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 }
 
 func setUpstreamProtocol(cluster *v2.Cluster, port *model.Port) {
-	if port.Protocol.IsHTTP2() {
+	switch {
+	case port.Protocol.IsHTTP():
```

Comment:
is HTTP is incorrect because it includes HTTP1, HTTP2 and gRPC as well, iirc. You need to check if its equal to model.ProtocolHTTP

Reply:
gonna tweak that now.